src/main/java/org/folio/rest/impl/AbstractInstanceRecordsAPI.java

```diff
@@ -99,7 +99,17 @@ protected void fetchRecordsByQuery(String sql, Supplier<Tuple> paramsSupplier, R
 
         RowStream<Row> rowStream = ar.result();
         rowStream
-        .exceptionHandler(e -> respondWithError(response, e, asyncResultHandler))
+        .exceptionHandler(e -> {
+          respondWithError(response, e, asyncResultHandler);
+          //database connections must be explicitly closed when an error is thrown 
+          //or it will be left in an open aborted state.
+          postgresClient.endTx(tx, h -> {
+            if (h.failed()) {
+              log.error("Unable to close database connection: " + h.cause().getMessage());
+              return;
+            }
+          });
```

Comment:
Does this mean that when the `exception handler` delegate is triggered the `end handler` delegate is not?

Reply:
That is behavior I observed.  I traced the actual 500 error being written out to this code.  When I added code to it that explicitly closed the database connection, I stopped seeing aborted connections in PG admin.

Reply:
Thank you for confirming that.